sdk/core/src/policies/retry_policies/retry_policy.rs

```diff
@@ -23,7 +23,7 @@ pub trait RetryPolicy {
     /// Determine how long before the next retry should be attempted.
     fn sleep_duration(&self, retry_count: u32) -> Duration;
     /// A Future that will wait until the request can be retried.
-    async fn wait(&self, retry_count: u32) {
+    async fn wait(&self, retry_count: u32, _error: &Error) {
```

Comment:
I think the error should probably go first. We may also want to consider calling it `last_error` or provide documentation that this is the error that caused the retry.